wrappers/src/query_cs.cpp

```diff
@@ -391,4 +392,38 @@ REALM_EXPORT Results* query_create_sorted_results(Query * query_ptr, SharedRealm
     });
 }
 
+
+#pragma mark  PrimaryKey Searches
+Row* row_for_primarykey(Table* table_ptr, ObjectSchema* object_schema, std::function<size_t(Table*, size_t columnIndex)> finder, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        size_t columnIndex = realm::npos;
+        if (object_schema->primary_key.size() > 0)
+            columnIndex = table_ptr->get_column_index(object_schema->primary_key);
+        if (columnIndex == realm::npos)
+            throw PrimaryKeyNotDeclaredException(object_schema->name);
```

Comment:
These few lines could be written as:
```c++
auto primary_prop = object_schema.primary_key_property();
if (!primary_prop) {
    throw …;
}
```

Then you'd go on to use `primary_prop->table_column` instead of `columnIndex`.